src/databasetasks.cpp

```diff
@@ -85,6 +85,13 @@ void DatabaseTasks::runTask(const DatabaseTask& task)
 
 void DatabaseTasks::flush()
 {
+	taskLock.lock();
+	flushInternal();
+	taskLock.unlock();
+}
+
+void DatabaseTasks::flushInternal()
+{
 	while (!tasks.empty()) {
 		runTask(tasks.front());
```

Comment:
It's good practice: http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#cp43-minimize-time-spent-in-a-critical-section

A mutex prevents any other threads from accessing the object. If, for some reason that other thread is the dispatcher thread, you'd block it from execution. While it's unlikely here, it's better to just avoid the potential of this occuring.